apps/src/templates/studioHomepages/Courses.jsx

```diff
@@ -113,7 +115,7 @@ class Courses extends Component {
             {/* English, teacher.  (Also can be shown when signed out.) */}
             {isEnglish && isTeacher && (
               <div className={'announcements'}>
-                {specialAnnouncement && (
+                {specialAnnouncement && teacherApplications2021 && (
```

Comment:
This will show no banner to teachers until we flip the flag, yeah? Seems fine if we want this to go live next week anyway.

I am wondering if there's a way to do this so we don't have to change this next time marketing wants to change the announcement on the teacher dashboard? I could imagine this prop (and DCDO flag) being `hideTeacherDashboardAnnouncement` or something, then we just switch it to true when we want this to be shown, and then marketing is free to update the announcement via `announcements.json` without our intervention?

This is sort of a half formed thought, as I'm not sure it would work with all the other places that you're using that teacher application DCDO flag, but something to consider.

Another thought is whether this could be `teacherApplicationsActive` or something, or does it feel like this is all 2021 specific?

Reply:
Yes, banners will be hidden until we flip the flag, `teacherApplications2021` is powered by the DCDO flag `teacher_applications_2021`, and the banner is still populated by `announcements.json` (was that your question? or did you mean that the teacher banner should be behind a different flag?). 

I'm open to changing the flag name! Clare said applications are already open this is just the marketing push, so maybe I should name it `teacherApplicationsActivePush`

Reply:
i agree that it seems like this flag could be made more generic/reusable, but probably better to wait and fully form the thought with some input from marketing. @clareconstantine do you know if this is something that's been discussed?

Reply:
Yeah sorry poorly worded, I think I meant the latter (I_or did you mean that the teacher banner should be behind a different flag?_) -- like in this context `teacherApplications2021` is just being used to say whether to show or hide the special announcement, which currently is the teacher application announcement. But, if announcement.json were changed (which marketing can do directly via Dropbox), some other announcement would be shown, and we have this very specific hide/show flag tied (at least in name) to teacher applications.

Anyway, I don't think this is a huge deal, but something to consider -- I do like that variable name update, but also not a dealbreaker.

Reply:
Ahh - I understand, have the flag as part of `announcement.json` instead of using DCDO, I like that idea a lot. I'll leave it as-is for now since there are other banners that aren't related to `announcements.json` which need to be hidden at the same time, so using DCDO will show/hide all at once. But that's a great idea for giving marketing more control!

Reply:
Sorry I think that's not what I'm saying either :) Piggybacking off of @cforkish comments and thinking a little bit more, I actually think what you could do here is get rid of the `teacherApplications2021` prop here entirely, and use the specialAnnouncement system as it's designed (I think it's actually designed to handle this case -- [here are some docs we put together if you haven't seen them](https://docs.google.com/document/d/1M1ON0pvvlI2MWJS43C5JbZERKLo3aUlQln4IFPtDrGk/edit#heading=h.e6zavcvgdze0), the bottom of the second page has some relevant information).

You can keep all your changes for the config for the 2021 banner in `announcements.json`, just remove the [line](https://github.com/code-dot-org/code-dot-org/blob/215ee1fc845233966f59ad68c07d83fabe5fb594/pegasus/sites.v3/code.org/announcements.json#L4) here until we actually want the banner to show up, then have someone in marketing add it on the day we want the banner to show up.

All that said, what you have here clearly works, and I understand this would mean having to do two things to get everything to show up for this (set dcdo flag and make dropbox edit) instead of one. Hopefully informative at least though for future reference!

Reply:
I've pushed an update the remove the DCDO flag for the announcements, and we'll just the json file, thanks for the help!!